public/app/features/dashboard/components/PanelEditor/state/actions.ts

```diff
@@ -45,7 +45,12 @@ export function panelEditorCleanUp(): ThunkResult<void> {
       modifiedSaveModel.id = sourcePanel.id;
 
       sourcePanel.restoreModel(modifiedSaveModel);
-      sourcePanel.getQueryRunner().pipeDataToSubject(panel.getQueryRunner().getLastResult());
+
+      // Resend last query result on source panel query runner
+      // But do this after the panel edit editor exit process has completed
+      setTimeout(() => {
```

Comment:
Not a huge fan of set timeout. Wouldn't dispatching a custom action do the same thing here?

Reply:
I tried moving this to a tunk action that is run last in this function but does not work, it is still executed before I need it to (before store change triggers with panel edit state closed), so I will need a setTimeout in the action so think this is the cleanest  way